### PR TITLE
Adds Association abstract class

### PR DESCRIPTION
**Pull request by nateshim-indico**

### CODE REVIEW

File: IndicoToolkit/Association/Association.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+        public bool isManuallyAddedPred(Prediction prediction)
+        {
+            return Extractions.isManuallyAddedPrediction(prediction);
+        }
+
+        /// <summary>
+        /// Boolean return value indicates whether or not sequences overlap.
+        /// </summary>
+        public bool sequencesOverlap(Prediction x, Prediction y)
```
Comment (nateshim-indico) [inline IndicoToolkit/Association/Association.cs:58]: C# does not allow non-member functions, so opted to enable the Association class to be initialized without providing any preds.

Comment (Scott771) [reply IndicoToolkit/Association/Association.cs:58]: OK

**Comment by nateshim-indico** [comment]
Yeah, it should be on a 'module' level, but unfortunately C# does not allow non-member functions, so I opted instead to enable the Association class to be initialized without providing any preds.

**Review by andrew8bit — APPROVED** [review]
Ah, interesting! Thanks for that - feel free to merge

**Pull request merged**